### ISSUE TITLE
shelljs.exec cannot find app.asar/node_modules/shelljs/src/exec-child.js

### ISSUE THREAD

**Issue by davidv1992** [shelljs/shelljs#881]
### Node version (or tell us if you're using electron or some other framework):
electron 1.8.6 prebuilt from electron-builder
### ShellJS version (the most recent version/Github branch you see the bug on):
shelljs 0.8.2
### Operating system:
ubuntu 18.04
### Description of the bug:
I am building an electron app using shelljs, and after packaging with electron-builder, using shelljs.exec fails, returning null and printing the following stacktrace on the console:

```
Error: Cannot find module '/redacted/dist/linux-unpacked/resources/app.asar/node_modules/shelljs/src/exec-child.js'
    at Function.Module._resolveFilename (module.js:547:15)
    at Function.Module._load (module.js:474:25)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:612:3
```

### Example ShellJS command to reproduce the error:

```javascript
exec(`find /bin /usr/bin /usr/local/bin -executable -name 'thunderbird' \\( -type f -or -type l \\)`)
```

**Comment by nfischer** [comment]
This *specific* message is unexpected. `exec-child.js` is a file we ship (you can see it in this repository). Can you check if this file exists anywhere in your node_modules/ folder?

```sh
$ find node_modules/ -name exec-child.js
```

---

That being said, we have never supported electron. We list some [known issues](https://github.com/shelljs/shelljs/wiki/Electron-compatibility) with possible workarounds. If there's an obvious fix for an electron-related issue, we're happy to consider it, but maintenance has only been best-effort, not guaranteed.

**Comment by davidv1992** [comment]
Figured out the issue. Because of the workaround you listed, it is using the system node.js, which doesnt know about asar archives. I am fixing this by migrating back to nodes native child_process.exec/execSync

**Comment by nfischer** [comment]
I see. In this case, this is likely not fixable in `shell.exec()`, but I believe this will be fixed in #866 (this does not rely on spawning something with the path to the `node` binary).
